### PR TITLE
Benchmark VRF and KES. Enable benchmarks in Github Actions CI

### PR DESCRIPTION
**Pull request by newhoggy**
This is a rebase of an earlier PR https://github.com/input-output-hk/cardano-base/pull/133 with updates to build with latest code.

Benchmark VRF and KES.  Enable benchmarks in Github Actions CI.

### CODE REVIEW

**Comment by newhoggy** [comment]
@tdammers I will need your help because after the rebase the benchmarks fail:

```
benchmarking KES/Sum6KES/genKey
Benchmark bench-crypto: ERROR
cabal: Benchmarks failed for bench:bench-crypto from
cardano-crypto-tests-2.0.0.
```

**Comment by newhoggy** [comment]
The failure was caused by the benchmarks not calling `sodiumInit` at start up.  This has been fixed.

File: cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs [hunk 1]
```diff
@@ -132,16 +133,25 @@ newtype SignKey = SignKey { unSignKey :: ForeignPtr SignKeyValue }
   deriving (Generic)
   deriving NoThunks via OnlyCheckWhnf SignKey
 
+instance NFData SignKey where
+  rnf a = seq a ()
+
```
Comment (erikd) [inline cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs:138]: Is it not possible to do?:
```
deriving instance NFData SignKey
```

**Review by erikd — APPROVED** [review]
Possibly worth removing the explicit `NFData` definitions.

File: cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-crypto-praos/src/Cardano/Crypto/VRF/Praos.hs:138]: This is the error:

```
 No instance for (NFData (ForeignPtr SignKeyValue))
```

**Pull request merged**